File: samples/csharp_dotnetcore/54.teams-task-module/Models/CardTaskFetchValue.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

using Newtonsoft.Json;

namespace Microsoft.BotBuilderSamples.Models
{
    public class CardTaskFetchValue<T>
    {
        [JsonProperty("type")]
        public object Type { get; set; } = "task/fetch";

        [JsonProperty("data")]
        public T Data { get; set; }
    }
}

File: samples/java_springboot/54.teams-task-module/src/main/java/com/microsoft/bot/sample/teamstaskmodule/models/CardTaskFetchValue.java
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

package com.microsoft.bot.sample.teamstaskmodule.models;

import com.fasterxml.jackson.annotation.JsonProperty;

public class CardTaskFetchValue<T> {
    @JsonProperty(value = "type")
    private Object type = "task/fetch";

    @JsonProperty(value = "data")
    private T data;

    public Object getType() {
        return type;
    }

    public void setType(Object type) {
        this.type = type;
    }

    public T getData() {
        return data;
    }

    public void setData(T data) {
        this.data = data;
    }
}
